src/components/managed-street.js

```diff
@@ -53,6 +53,92 @@ AFRAME.registerComponent('managed-street', {
     // Bind the method to preserve context
     this.refreshFromSource = this.refreshFromSource.bind(this);
   },
+  /**
+   * Inserts a new street segment at the specified index
+   * @param {number} index - The index at which to insert the new segment
+   * @param {string} type - The segment type (e.g., 'drive-lane', 'bike-lane')
+   * @param {Object} [segmentObject] - Optional configuration object for the segment
+   * @returns {Element} The created segment element
+   */
+  insertSegment: function (index, type, segmentObject = null) {
+    // Validate index
+    if (index < 0 || index > this.managedEntities.length) {
+      console.error('[managed-street] Invalid index for insertion:', index);
+      return;
+    }
+
+    // Create new segment entity
+    const segmentEl = document.createElement('a-entity');
+
+    // Get default properties for this segment type from STREET.types
+    const defaultProps = window.STREET.types[type] || {};
+
+    // Set up basic segment properties, merging defaults with any provided custom properties
+    const segmentProps = {
+      type: type,
+      width: segmentObject?.width || defaultProps.width || 3,
+      length: this.data.length,
+      level: segmentObject?.level ?? defaultProps.level ?? 0,
+      direction:
+        segmentObject?.direction || defaultProps.direction || 'outbound',
+      color:
+        segmentObject?.color ||
+        defaultProps.color ||
+        window.STREET.colors.white,
+      surface: segmentObject?.surface || defaultProps.surface || 'asphalt'
+    };
+
+    // Set the segment component with properties
+    segmentEl.setAttribute('street-segment', segmentProps);
+
+    // Set the layer name for the segment
+    const layerName = segmentObject?.name || `${type} • default`;
+    segmentEl.setAttribute('data-layer-name', layerName);
+
+    // If custom segment object is provided, wait for segment to load then generate its components
+    if (segmentObject) {
+      segmentEl.addEventListener('loaded', () => {
+        // Use the generateComponentsFromSegmentObject method from street-segment component
+        const streetSegmentComponent = segmentEl.components['street-segment'];
+        if (streetSegmentComponent) {
+          streetSegmentComponent.generateComponentsFromSegmentObject(
+            segmentObject
+          );
+        }
+      });
+    }
+
+    // Insert the segment at the specified index in the DOM
+    if (index === this.managedEntities.length) {
+      this.el.appendChild(segmentEl);
+    } else {
+      const referenceNode = this.managedEntities[index];
+      this.el.insertBefore(segmentEl, referenceNode);
+    }
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore#referencenode
_referenceNode_
The node before which newNode is inserted. If this is null, then newNode is inserted at the end of node's child nodes.
So instead of 
```js
    if (index === this.managedEntities.length) {
      this.el.appendChild(segmentEl);
    } else {
      const referenceNode = this.managedEntities[index];
      this.el.insertBefore(segmentEl, referenceNode);
    }
```
you can do:
```js
    const referenceNode = this.managedEntities[index] ?? null;
    this.el.insertBefore(segmentEl, referenceNode);
```

This is what we do already in
https://github.com/3DStreet/3dstreet/blob/172d36051e011365c21fc8a22310b47bd6d88ed2/src/editor/lib/commands/EntityRemoveCommand.js#L37-L39

Reply:
1781d2b3baafc7d9c2d5ad919a2e0bab5acc04c7